src/main/java/org/springframework/social/openidconnect/HttpClientFactory.java

```diff
@@ -114,4 +116,26 @@ private static X509HostnameVerifier getVerifier(boolean isStrict){
         return SSLSocketFactory.STRICT_HOSTNAME_VERIFIER;
     }
 
+    /**
+     * Gets the TLS version from jdk parameter.  Defaults to version 1.2 for JDK 1.7 if not provided
+     * @return
+     */
+    public static String getTlsVersion() {
+        // don't use "jdk.tls.client.protocols" as it was causing unexpected exception during unit testing
+        String tlsVer = System.getProperty("tlsProtocol");
+
+        if(StringUtils.isEmpty(tlsVer)) {
+            double javaVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+            if(javaVersion == 1.7) {
```

Comment:
Abhi - if we did default to TLSv1.2 and a JDK 1.6 app calls this API without specifying the TLS version, wouldn't it fail? If so, won't that break anyone at JDK 1.6 or lower until they made a coding change? I prefer the idea of handling JDK 1.7 explicitly and being backward and forward compatible for all other JDKs.